### PR TITLE
Don't crop rotated frames in ReferenceCamera

### LINKED ISSUE #228: Camera rotation crops frame
In ReferenceCamera#transformImage we apply several user defined transformations, one of which is rotation. The current rotation system crops the frame, especially in cases where the aspect ratio is not 1:1. There is a TODO in the code to correct this, and there is a good example at: http://stackoverflow.com/questions/22041699/rotate-an-image-without-cropping-in-opencv-in-c

* Change the transformImage code so that it adjusts the size of the image before rotating so that the full rotated image will fit in the new frame.

### PR DESCRIPTION
**Pull request by richard-sim**
In ReferenceCamera#transformImage, adjust the size of the image before rotating so that the full rotated image will fit in the new frame. Fixes #228

### CODE REVIEW

File: src/main/java/org/openpnp/machine/reference/ReferenceCamera.java [hunk 1]
```diff
@@ -206,14 +195,69 @@ protected BufferedImage transformImage(BufferedImage image) {
...
+                put(0, 0, 1, 0, offsetX);
+                put(1, 0, 0, 1, offsetY);
+            }
+        };
+        
+        Mat dst = mat.clone();
+        Imgproc.warpAffine(mat, dst, mapMatrix, mat.size(), Imgproc.INTER_LINEAR);
+
+        mapMatrix.release();
+
```
Comment (vonnieda) [inline src/main/java/org/openpnp/machine/reference/ReferenceCamera.java]: I think we're missing a mat.release() here?

**Comment by richard-sim** [comment]
Erk, good catch - thanks!

**Comment by vonnieda** [comment]
This all looks great aside from that one mat.release comment. If you'll fix that I'll get these merged.

**Comment by richard-sim** [comment]
Should be good now - I also found another missed release() while I was at it. Double oops.

**Comment by vonnieda** [comment]
Awesome work @richard-sim, thank you!

**Pull request merged**